lib/galaxy/tool_util/loader_directory.py

```diff
@@ -235,8 +236,8 @@ def looks_like_a_cwl_artifact(path, classes=None):
     if not _has_extension(path, CWL_EXTENSIONS):
         return False
 
-    as_dict = as_dict_if_looks_like_yaml_or_cwl_with_class(path, classes)
-    if as_dict is None:
+    is_yaml, as_dict = as_dict_if_looks_like_yaml_or_cwl_with_class(path, classes)
+    if not is_yaml is None:
```

Comment:
```suggestion
    if not is_yaml:
```
since this is now a proper boolean